rotkehlchen/api/rest.py

```diff
@@ -4037,3 +4042,166 @@ def query_kraken_staking_events(
         status_code = _get_status_code_from_async_response(response)
         result_dict = {'result': response['result'], 'message': response['message']}
         return api_response(process_result(result_dict), status_code=status_code)
+
+    def _get_user_added_assets(self) -> Dict[str, Any]:
+        try:
+            assets = GlobalDBHandler().get_user_added_assets(user_db=self.rotkehlchen.data.db)
+            dirpath = Path(mkdtemp())
+            dirpath.mkdir(parents=True, exist_ok=True)
+            serialized = []
+            for asset_identifier in assets:
+                try:
+                    asset = Asset(asset_identifier)
+                    serialized.append(asset_to_dict(asset))
+                except UnknownAsset as e:
+                    log.error(e)
+        except PermissionError as e:
+            return wrap_in_fail_result(
+                message=f'Failed to create information file. {str(e)}',
+                status_code=HTTPStatus.INTERNAL_SERVER_ERROR,
+            )
+        except sqlcipher.Error as e:  # pylint: disable=no-member
+            return wrap_in_fail_result(
+                message=f'Failed to retrieve the list of assets {str(e)}',
+                status_code=HTTPStatus.INTERNAL_SERVER_ERROR,
+            )
+        cursor = GlobalDBHandler()._conn.cursor()
+        query = cursor.execute('SELECT value from clean_db.settings WHERE name=="version";')
+        version = query.fetchone()[0]
+        data = {
+            'version': version,
+            'assets': serialized,
+        }
+
+        zip_path = dirpath / 'assets.zip'
+        with ZipFile(zip_path, 'w') as assets_zip:
+            assets_zip.writestr(
+                zinfo_or_arcname='assets.json',
+                data=json.dumps(data),
+            )
+
+        return _wrap_in_ok_result({'file': str(zip_path)})
+
+    @require_loggedin_user()
+    def get_user_added_assets(self, async_query: bool) -> Response:
+        if async_query:
+            return self._query_async(
+                command='_get_user_added_assets',
+            )
+
+        response = self._get_user_added_assets()
+        result = response['result']
+        msg = response['message']
+        status_code = _get_status_code_from_async_response(response)
+        if result is None:
+            return api_response(wrap_in_fail_result(msg), status_code=status_code)
+
+        # success
+        result_dict = _wrap_in_result(result, msg)
+        return api_response(result_dict, status_code=status_code)
+
```

Comment:
IMPORTANT NOTE: I don't know exactly where ... probably somewhere in the globalDB handler? A new class there? I can't tell you off the top of my head without thinking more but this is not the place to add all this logic.

I have told you before do not dump so much logic in api/rest.py. It's getting close to 5k lines!!!

Reply:
You told me after this PR. Moving it <img src="https://media1.giphy.com/media/Xg9wF4Uq0VrjE1WpNQ/giphy.gif"/>